src/fontcache.cpp

```diff
@@ -960,7 +963,51 @@ static void LoadWin32Font(FontSize fs)
 	if (StrEmpty(settings->font)) return;
 
 	LOGFONT logfont = { 0 };
-	if (settings->os_handle != nullptr) logfont = *(const LPLOGFONT)settings->os_handle;
+	logfont.lfPitchAndFamily = fs == FS_MONO ? FIXED_PITCH : VARIABLE_PITCH;
+	logfont.lfCharSet = DEFAULT_CHARSET;
+	logfont.lfOutPrecision = OUT_OUTLINE_PRECIS;
+	logfont.lfClipPrecision = CLIP_DEFAULT_PRECIS;
+
+	if (settings->os_handle != nullptr) {
+		logfont = *(const LPLOGFONT)settings->os_handle;
+	} else if (strchr(settings->font, '.') != nullptr && FileExists(settings->font)) {
+		/* Might be a font file name, try load it. */
+		TCHAR fontPath[MAX_PATH];
+		convert_to_fs(settings->font, fontPath, lengthof(fontPath), false);
+
+		if (AddFontResourceEx(fontPath, FR_PRIVATE, 0) != 0) {
+			/* Try a nice little undocumented function first for getting the internal font name.
+			 * Some documentation is fount at: http://www.undocprint.org/winspool/getfontresourceinfo */
+			typedef BOOL(WINAPI * PFNGETFONTRESOURCEINFO)(LPCTSTR, LPDWORD, LPVOID, DWORD);
+#ifdef UNICODE
+			static PFNGETFONTRESOURCEINFO GetFontResourceInfo = (PFNGETFONTRESOURCEINFO)GetProcAddress(GetModuleHandle(_T("Gdi32")), "GetFontResourceInfoW");
+#else
+			static PFNGETFONTRESOURCEINFO GetFontResourceInfo = (PFNGETFONTRESOURCEINFO)GetProcAddress(GetModuleHandle(_T("Gdi32")), "GetFontResourceInfoA");
+#endif
+
+			if (GetFontResourceInfo != nullptr) {
+				/* Try to query an array of LOGFONTs that describe the file. */
+				DWORD len = 0;
+				if (GetFontResourceInfo(fontPath, &len, nullptr, 2)) {
+					LOGFONT *buf = (LOGFONT *)AllocaM(byte, len);
```

Comment:
I guess it's not necessary to protect against len==0 here, since the AddFontResourceEx would have failed then?

Reply:
Good point. Can't be too careful with undocumented calls.